sootup.core/src/main/java/sootup/core/jimple/basic/Local.java

```diff
@@ -153,6 +154,36 @@ public List<Stmt> getDefsForLocalUse(StmtGraph<?> graph, Stmt stmt) {
     return defStmts;
   }
 
+  public List<Stmt> getLocalOccurrences(Collection<Stmt> stmts) {
+    List<Stmt> localOccurrences = new ArrayList<>();
+    for (Stmt stmt : stmts) {
+      if (stmt instanceof AbstractDefinitionStmt) {
```

Comment:
Is it sufficient to only check `AbstractDefinitionStmt` and `JIfStmt`?

Reply:
Nope, but I just saw that for these two common cases this occurs, we can later add if we find any other cases where assignments would lead to removal of variables

Reply:
with the method name, shouldnt we move it into the body as well (its like the method in the conversation above)